pyproject.toml

```diff
@@ -1,70 +1,109 @@
-requires-python = ">=3.7"
+requires-python = ">=3.7, <3.13"
 license = {text = "MIT License"}
 classifiers = [
-    "Development Status :: 3 - Alpha",
-    "Intended Audience :: Developers",
-    "Topic :: Scientific/Engineering",
-    "License :: OSI Approved :: MIT License",
-    "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.7",
-    "Programming Language :: Python :: 3.8",
-    "Programming Language :: Python :: 3.9",
-    "Programming Language :: Python :: 3.10",
-    "Programming Language :: Python :: 3.11",
+  "Development Status :: 3 - Alpha",
+  "Intended Audience :: Developers",
+  "Topic :: Scientific/Engineering",
+  "License :: OSI Approved :: MIT License",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12"
 ]
 dependencies = [
-    "av",
-    "fiftyone",
-    "matplotlib",
-    "nibabel",
-    "nilearn",
-    "numpy",
-    "Pillow",
-    "scipy",
-    "scikit-learn",
+  "av>=12.0.0",
+  "bids2table>=0.1.0b0",
+  "elbow>=0.1.1",
+  "fiftyone>=0.23.7",
+  "matplotlib>=3.7.1",
+  "nibabel>=5.2.1",
+  "nilearn>=0.10.3",
+  "numpy>=1.24.3",
+  "pandas>=2.0.3",
+  "pillow>=10.3.0",
+  "scikit-learn>=1.3.0",
+  "scipy>=1.10.1"
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
-test = [
-    "pytest",
-    "pytest-cov",
-]
-dev = [
-    "black==23.3.0",
-    "flake8==5.0.4",
-    "isort==5.11.5",
-    "pre-commit==3.3.1",
-    "setuptools-scm",
-]
+test = ["pytest>=8.0.2", "pytest-cov>=4.1.0"]
+dev = ["mypy>=1.9.0", "pre-commit>=3.7.0", "ruff>=0.3.5", "setuptools-scm"]
 
 [project.urls]
-"Homepage" = "https://github.com/cmi-dair/niftyone"
+"Homepage" = "https://github.com/childmindresearch/niftyone"
 
 [project.scripts]
 niftyone = "niftyone.__main__:main"
 
 [tool.setuptools_scm]
 write_to = "src/niftyone/_version.py"
 
-[tool.black]
-line-length = 88
-target_version = ['py37']
-
-[tool.isort]
-profile = "black"
-
 [tool.pytest.ini_options]
+pythonpath = ["src"]
+testpaths = ["tests"]
 log_cli = true
 log_cli_level = "DEBUG"
+
+[tool.ruff]
+exclude = [
+  ".bzr",
+  ".direnv",
+  ".eggs",
+  ".git",
+  ".git-rewrite",
+  ".hg",
+  ".mypy_cache",
+  ".nox",
+  ".pants.d",
+  ".pytype",
+  ".ruff_cache",
+  ".svn",
+  ".tox",
+  ".venv",
+  "__pypackages__",
+  "_build",
+  "buck-out",
+  "build",
+  "dist",
+  "node_modules",
+  "venv"
+]
+line-length = 88
+indent-width = 4
+src = ["src"]
+target-version = "py38"
+
+[tool.ruff.lint]
+select = ["ANN", "D", "E", "F", "I"]
+ignore = [
+  "ANN101",  # self should not be annotated.
+  "ANN102"  # cls should not be annotated.
+]
+fixable = ["ALL"]
+unfixable = []
+dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
+
+[tool.ruff.lint.pydocstyle]
+convention = "google"
+
+[tool.ruff.lint.per-file-ignores]
+"__init__.py" = ["F401"]
```

Comment:
Thanks, this helps get rid of all the `noqa`s :) I recently realized you could do this.